include/pybind11/detail/smart_holder_type_casters.h

```diff
@@ -694,17 +694,22 @@ struct smart_holder_type_caster_load {
     holder_type &holder() const { return load_impl.loaded_v_h.holder<holder_type>(); }
 
     // have_holder() must be true or this function will fail.
-    void throw_if_uninitialized_or_disowned_holder() const {
+    void throw_if_uninitialized_or_disowned_holder(const char *typeid_name) const {
+        static const std::string missing_value_msg = "Missing value for wrapped C++ type `";
         if (!holder().is_populated) {
-            pybind11_fail("Missing value for wrapped C++ type:"
-                          " Python instance is uninitialized.");
+            throw value_error(missing_value_msg + clean_type_id(typeid_name)
```

Comment:
I dont' suppose you can use f-strings?

Reply:
Interesting idea, but no. We don't have f-strings in C++ (I never thought about that before, not sure how feasible that is), and compared to relying on `std::string` `operator+`, the overhead of using the Python C API is not something I'd want to have in an error handling code path like here.